### PR TITLE
Verify merge master -> feature/firebase-service

### PR DESCRIPTION
**Pull request by djsanabriac**

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -22,9 +22,14 @@
             "main": "src/main.ts",
             "polyfills": "src/polyfills.ts",
             "tsConfig": "tsconfig.app.json",
-            "aot": false,
-            "assets": ["src/favicon.ico", "src/assets"],
-            "styles": ["src/styles.scss", "src/app/shared/theme/theme.scss"],
```
Comment (JuanSeBestia) [inline angular.json:27]: Please add "src/app/shared/theme/theme.scss"

File: src/app/app.module.ts [hunk 2]
```diff
@@ -48,7 +50,7 @@ import { ReactiveFormsModule, FormsModule } from '@angular/forms';
   imports: [
     BrowserModule,
     AppRoutingModule,
-    AngularFireModule,
+    AngularFireModule.initializeApp(environment.firebase),
```
Comment (JuanSeBestia) [inline src/app/app.module.ts:53]: Please, not override the config for initialization. look src/main.ts and https://github.com/angular/angularfire/issues/2360#issuecomment-607916972

File: src/app/shared/models/courier.model.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+export class Courier implements Deserializable{
+
+    id: string;
+    name: string;
+    idNumber: string;
+    plate: string;
+    vehicle: string;
+    company: string;
+
+    deserialize(input: any) {
```
Comment (JuanSeBestia) [inline src/app/shared/models/courier.model.ts:12]: `any` never please use Partial<Courier> instead

File: src/app/shared/models/deserializable.model.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export interface Deserializable {
```
Comment (JuanSeBestia) [inline src/app/shared/models/deserializable.model.ts:1]: FYI: In TypeScript exist Class Abstract https://www.typescriptlang.org/docs/handbook/classes.html#abstract-classes.

This comment is not a required change, don't worry, keep with this code :)

Comment (JuanSeBestia) [reply src/app/shared/models/deserializable.model.ts:1]: O maybe yes, because all the classes have the same implementation for deserialize

Comment (JuanSeBestia) [reply src/app/shared/models/deserializable.model.ts:1]: Please contact me to review that

File: src/app/shared/services/firebase.service.ts [hunk 5]
```diff
@@ -1,9 +1,231 @@
...
+  updateOrder(order: Order){
+    return this.firestore.doc("orders/"+order.id).update({...order});
+  }
+
+  deleteOrder(orderId: string){
+    return this.firestore.doc("orders/" + orderId).delete();
+  }
+
+  // --------------------------------------------------------------------------
+
```
Comment (JuanSeBestia) [inline src/app/shared/services/firebase.service.ts:230]: Too long file, please split in orders.service.ts, patient.service.ts, couriers.service.ts

File: src/app/shared/services/firebase.service.ts [hunk 6]
```diff
@@ -1,9 +1,231 @@
...
+          console.debug("[DEBUG] Error deleting", reason);
+        });
+
+  }
+
+  getUsers(){
+    this.firestore.collection<User>('users').valueChanges().subscribe((data)=>{
+      this.users = data;
+      console.debug(this.users);
+    });
```
Comment (JuanSeBestia) [inline src/app/shared/services/firebase.service.ts:80]: Fort curiosity, why need to keep listening to the changes for Users

**Comment by JuanSeBestia** [comment]
Amazing work :)

File: angular.json (hunk 1 above)
Comment (djsanabriac) [reply angular.json:27]: Done

File: src/app/app.module.ts (hunk 2 above)
Comment (djsanabriac) [reply src/app/app.module.ts:53]: Done

File: src/app/shared/models/deserializable.model.ts (hunk 4 above)
Comment (djsanabriac) [reply src/app/shared/models/deserializable.model.ts:1]: You're damn right, done.

File: src/app/shared/models/courier.model.ts (hunk 3 above)
Comment (djsanabriac) [reply src/app/shared/models/courier.model.ts:12]: Changed

File: src/app/shared/services/firebase.service.ts (hunk 5 above)
Comment (djsanabriac) [reply src/app/shared/services/firebase.service.ts:230]: Done

File: src/app/shared/services/firebase.service.ts (hunk 6 above)
Comment (djsanabriac) [reply src/app/shared/services/firebase.service.ts:80]: It's the way Firebase is used

**Comment by JuanSeBestia** [comment]
@djsanabriac Tienes conflictos :(

**Pull request merged**